File: binding/src/multimedia/libqaudiobufferinput.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qaudiobufferinput.html
pub const qaudiobufferinput = struct {

    /// New constructs a new QAudioBufferInput object.
    pub fn New() QtC.QAudioBufferInput {
        return qtc.QAudioBufferInput_new();
    }


    /// New2 constructs a new QAudioBufferInput object.
    pub fn New2(format: ?*anyopaque) QtC.QAudioBufferInput {
        return qtc.QAudioBufferInput_new2(@ptrCast(format));
    }


    /// New3 constructs a new QAudioBufferInput object.
    pub fn New3(parent: ?*anyopaque) QtC.QAudioBufferInput {
        return qtc.QAudioBufferInput_new3(@ptrCast(parent));
    }


    /// New4 constructs a new QAudioBufferInput object.
    pub fn New4(format: ?*anyopaque, parent: ?*anyopaque) QtC.QAudioBufferInput {
        return qtc.QAudioBufferInput_new4(@ptrCast(format), @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAudioBufferInput_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qaudiobufferinput.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SendAudioBuffer(self: ?*anyopaque, audioBuffer: ?*anyopaque) bool {
        return qtc.QAudioBufferInput_SendAudioBuffer(@ptrCast(self), @ptrCast(audioBuffer));
    }

    pub fn Format(self: ?*anyopaque, ) QtC.QAudioFormat {
        return qtc.QAudioBufferInput_Format(@ptrCast(self));
    }

    pub fn CaptureSession(self: ?*anyopaque, ) QtC.QMediaCaptureSession {
        return qtc.QAudioBufferInput_CaptureSession(@ptrCast(self));
    }

    pub fn ReadyToSendAudioBuffer(self: ?*anyopaque, ) void {
        qtc.QAudioBufferInput_ReadyToSendAudioBuffer(@ptrCast(self));
    }

    pub fn OnReadyToSendAudioBuffer(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QAudioBufferInput_Connect_ReadyToSendAudioBuffer(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAudioBufferInput_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qaudiobufferinput.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAudioBufferInput_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qaudiobufferinput.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QAudioBufferInput_Delete(@ptrCast(self));
    }
};

File: binding/src/multimedia/libqaudiobufferinput.cpp
#include <QAudioBuffer>
#include <QAudioBufferInput>
#include <QAudioFormat>
#include <QMediaCaptureSession>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qaudiobufferinput.h>
#include "libqaudiobufferinput.h"
#include "libqaudiobufferinput.hxx"

QAudioBufferInput* QAudioBufferInput_new() {
	 return new QAudioBufferInput();
}

QAudioBufferInput* QAudioBufferInput_new2(const QAudioFormat* format) {
	 return new QAudioBufferInput(*format);
}

QAudioBufferInput* QAudioBufferInput_new3(QObject* parent) {
	 return new QAudioBufferInput(parent);
}

QAudioBufferInput* QAudioBufferInput_new4(const QAudioFormat* format, QObject* parent) {
	 return new QAudioBufferInput(*format, parent);
}

libqt_string QAudioBufferInput_Tr(const char* s) {
	QString _ret = QAudioBufferInput::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

bool QAudioBufferInput_SendAudioBuffer(QAudioBufferInput* self, const QAudioBuffer* audioBuffer) {
	return self->sendAudioBuffer(*audioBuffer);
}

QAudioFormat* QAudioBufferInput_Format(const QAudioBufferInput* self) {
	return new QAudioFormat(self->format());
}

QMediaCaptureSession* QAudioBufferInput_CaptureSession(const QAudioBufferInput* self) {
	return self->captureSession();
}

void QAudioBufferInput_ReadyToSendAudioBuffer(QAudioBufferInput* self) {
	self->readyToSendAudioBuffer();
}

void QAudioBufferInput_Connect_ReadyToSendAudioBuffer(QAudioBufferInput* self, intptr_t slot) {
    void (*slotFunc)(QAudioBufferInput*) = reinterpret_cast<void (*)(QAudioBufferInput*)>(slot);
    QAudioBufferInput::connect(self, &QAudioBufferInput::readyToSendAudioBuffer, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QAudioBufferInput_Tr2(const char* s, const char* c) {
	QString _ret = QAudioBufferInput::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QAudioBufferInput_Tr3(const char* s, const char* c, int n) {
	QString _ret = QAudioBufferInput::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QAudioBufferInput_Delete(QAudioBufferInput* self) {
    delete self;
}

